doc/src/manual/multi-threading.md

```diff
@@ -213,3 +213,33 @@ therefore a blocking call like other Julia APIs.
 It is very important that the called function does not call back into Julia, as it will segfault.
 
 `@threadcall` may be removed/changed in future versions of Julia.
+
+## Caveats
+
+At this time, most operations in the Julia runtime and standard libraries
+are thread safe.
+However, in some areas work on stabilizing thread support is ongoing.
+Multi-threaded programming has many inherent difficulties, and if a program
+using threads exhibits unusual or undesirable behavior (e.g. crashes or
+mysterious results), thread interactions should typically be suspected first.
+
+There are a few specific limitations and warnings to be aware of when using
+threads in Julia:
+
+  * Base collection types require manual locking if used simultaneously by
+    multiple threads (common examples include `push!` on arrays, or inserting
+    items into a `Dict`).
+  * After a task starts running on a certain thread (e.g. via `@spawn`), it
+    will always be restarted on the same thread after blocking. In the future
+    this limitation will be removed, and tasks will migrate between threads.
```

Comment:
Can you avoid thread migration by `@spwan @async f()`? If so, maybe mention that this is the forward-compatible way to rely on this property? (Not sure if we should be encouraging this pattern, though.)

Reply:
Hm, curious. Probably not—instead seems like `@async` should inherit the sticky bit from the parent in the future.

Reply:
A function using `@async` internally may be called via `@spawn`.  If `@async` inherits the sticky bit, this internal use of `@async` becomes actually equivalent to `@spawn`, right?

I think it's possible to use `@async` in a way concurrent-safe but thread-unsafe.  If so, isn't it dangerous to inherit the sticky bit?